tests/test_core.py

```diff
@@ -359,81 +297,25 @@ def test_deletes(config, requests_mock):
-                "etag": "wrong etag"
-            }
-        }],
-    },
-     errors.BadRequest('errorMessage',
-                       details=[{
-                           'type': DatabricksError._error_info_type,
-                           'reason': 'error reason',
-                           'domain': 'error domain',
-                           'metadata': {
-                               'etag': 'error etag'
-                           },
-                       }])),
-    (401, {}, {
+@pytest.mark.parametrize(
+    'status_code,headers,body,expected_error',
+    [(401, {}, {
         'error_code': 'UNAUTHORIZED',
         'message': 'errorMessage',
     },
-     errors.Unauthenticated('errorMessage. Config: host=http://localhost, auth_type=noop',
-                            error_code='UNAUTHORIZED')),
-    (403, {}, {
-        'error_code': 'FORBIDDEN',
-        'message': 'errorMessage',
-    },
-     errors.PermissionDenied('errorMessage. Config: host=http://localhost, auth_type=noop',
-                             error_code='FORBIDDEN')),
-    (429, {}, {
-        'error_code': 'TOO_MANY_REQUESTS',
-        'message': 'errorMessage',
-    }, errors.TooManyRequests('errorMessage', error_code='TOO_MANY_REQUESTS', retry_after_secs=1)),
-    (429, {
-        'Retry-After': '100'
-    }, {
-        'error_code': 'TOO_MANY_REQUESTS',
-        'message': 'errorMessage',
-    }, errors.TooManyRequests('errorMessage', error_code='TOO_MANY_REQUESTS', retry_after_secs=100)),
-    (503, {}, {
-        'error_code': 'TEMPORARILY_UNAVAILABLE',
-        'message': 'errorMessage',
-    }, errors.TemporarilyUnavailable('errorMessage', error_code='TEMPORARILY_UNAVAILABLE',
-                                     retry_after_secs=1)),
-    (503, {
-        'Retry-After': '100'
-    }, {
-        'error_code': 'TEMPORARILY_UNAVAILABLE',
-        'message': 'errorMessage',
-    },
-     errors.TemporarilyUnavailable('errorMessage', error_code='TEMPORARILY_UNAVAILABLE',
-                                   retry_after_secs=100)),
-    (404, {}, {
-        'scimType': 'scim type',
-        'detail': 'detail',
-        'status': 'status',
-    }, errors.NotFound('scim type detail', error_code='SCIM_status')),
-])
+      errors.Unauthenticated('errorMessage. Config: host=http://localhost, auth_type=noop',
+                             error_code='UNAUTHORIZED')),
+     (403, {}, {
+         'error_code': 'FORBIDDEN',
+         'message': 'errorMessage',
+     },
+      errors.PermissionDenied('errorMessage. Config: host=http://localhost, auth_type=noop',
+                              error_code='FORBIDDEN')), ])
 def test_error(config, requests_mock, status_code, headers, body, expected_error):
     client = ApiClient(config)
     requests_mock.get("/test", json=body, status_code=status_code, headers=headers)
     with pytest.raises(DatabricksError) as raised:
-        client._perform("GET", "http://localhost/test", headers={"test": "test"})
```

Comment:
Note that after this PR, it isn't possible to call `ApiClient._perform()` anymore.